src/Plugin.php

```diff
@@ -42,6 +42,15 @@ public static function preInit() {
     add_filter('woocommerce_variable_price_html', __NAMESPACE__ . '\WooCommerce::woocommerce_variable_sale_price_html', 10, 2);
 
     add_action('parse_request', __CLASS__  . '::parse_request');
+
+    // We need woocommerce-german-market plugin being active to add product
+    // delivery time support to products filtering in the woocommerce layered
+    // navigation.
+    $active_plugins = apply_filters('active_plugins', get_option('active_plugins'));
+    if (in_array('woocommerce-german-market/WooCommerce-German-Market.php', $active_plugins)) {
+      // Registers products filter widgets supporting delivery time.
+      add_action('widgets_init', __CLASS__ . '::widgets_init');
```

Comment:
Every feature in core-standards and shop-standards should be located in an own class file (here: `DeliveryTime`), so that the different functionalities are properly separated.

We already started to do this and should continue to do so (also retroactively with other existing features).  If we do not, then whole plugin will become a hodgepodge of spaghetti code soon and it will be hard to cleanly remove one of the features in full in the future.

In this case I would suggest `DeliveryTime` and in case multiple classes are needed (which seems to be the case) then the following structure:
```
DeliveryTime\Widget (including widgets_init, query_vars, pre_get_posts)
DeliveryTime\WidgetLayeredNav
DeliveryTime\WidgetLayeredNavFilters
```

The only lines that stay in `Plugin` is the one commented on here.

Reply:
@sun I decided to use "ProductsFiltering" instead of "DeliveryTime". IMO the implemented feature is about products filtering, being delivery time just the current case, but different filtering criteria could be added in the future.